ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDReminderCell.m

```diff
@@ -17,14 +17,163 @@ @interface CDReminderCell () <UITextViewDelegate>
+		[self setConstraintPriorityForPriorityAttribute];
+	} else {
+		self.textView.textContainerInset = UIEdgeInsetsMake(0.0, 0.0, 0.0, 0.0);
+		self.priority.alpha = 0;
+	}
+
+	if (reminder.taskDate && (!reminder.note || [reminder.note isEqualToString:@""])) {
+		[self setConstraintPriorityForSimpleCellPlusDate];
+	} else if (reminder.taskDate && reminder.note) {
+		[self setConstraintPriorityForSimpleCellPlusDateAndNote];
+	} else if (!reminder.taskDate && reminder.note) {
+		[self setConstraintPriorityForSimpleCellPlusNote];
+	}
+	
+}
+
+#pragma mark - Helper Methods for changing priorities
+
+- (void)setConstraintPriorityForSimpleCell {
+	
+	self.textViewToCellBottom.priority = 999;
+	self.textViewToCellBottom.constant = 2;
+	
+	self.textViewToNoteBottomConstraint.priority = 800;
+	self.textViewToDateBottomConstraint.priority = 800;
+	self.dateToNote.priority = 800;
+	self.dateToCellBottom.priority = 800;
+	self.noteToBottomCell.priority = 800;
+	
+	self.date.alpha = 0;
+	self.note.alpha = 0;
+	
+	[self.contentView layoutIfNeeded];
 
 }
 
+- (void)setConstraintPriorityForSimpleCellPlusDate {
+	
+	self.textViewToDateBottomConstraint.priority = 999;
+	self.dateToCellBottom.priority = 999;
+	self.dateToCellBottom.constant = 2;
+	self.date.alpha = 100;
+	self.note.alpha = 0;
+	
+	self.textViewToNoteBottomConstraint.priority = 800;
+	self.textViewToCellBottom.priority = 800;
+	self.dateToNote.priority = 800;
+	self.noteToBottomCell.priority = 800;
+	
+	[self.contentView layoutIfNeeded];
+	
+}
+
+- (void)setConstraintPriorityForSimpleCellPlusNote {
+
+	self.textViewToNoteBottomConstraint.priority = 999;
+	self.textViewToNoteBottomConstraint.constant = 2;
+	self.noteToBottomCell.priority = 999;
+	self.noteToBottomCell.constant = 2;
+	self.note.alpha = 100;
+	self.date.alpha = 0;
+	
+	self.textViewToDateBottomConstraint.priority = 800;
+	self.dateToNote.priority = 800;
+	self.dateToCellBottom.priority = 800;
+	self.textViewToCellBottom.priority = 800;
+	
+	[self.contentView layoutIfNeeded];
+
+}
+
+- (void)setConstraintPriorityForSimpleCellPlusDateAndNote {
+	
+	self.textViewToDateBottomConstraint.priority = 999;
+	self.dateToNote.priority = 999;
+	self.noteToBottomCell.priority = 999;
+	self.noteToBottomCell.constant = 2;
+	
+	self.note.alpha = 100;
+	self.date.alpha = 100;
+	
+	self.textViewToNoteBottomConstraint.priority = 800;
+	self.textViewToCellBottom.priority = 800;
+	self.dateToCellBottom.priority = 800;
+	
+	[self.contentView layoutIfNeeded];
+	
+}
+
+- (void)setConstraintPriorityForPriorityAttribute {
+	
+	self.textView.textContainerInset = UIEdgeInsetsMake(self.priority.bounds.size.height, 0.0, 0.0, 0.0);
+	self.priority.alpha = 100;
+	
+	[self.contentView layoutIfNeeded];
+	
+}
+
+#pragma mark - Helper method for converting Priority attribute into string
+
+- (NSString *)stringFromNSNumberPriority:(NSNumber *)priorityNumber {
+	
+	switch (priorityNumber.integerValue) {
```

Comment:
In Reminder "@property (nullable, nonatomic, retain) NSNumber *priority;" should be a scalar(Int_16) and you should create a NS_ENUM for this in that class. Priority Low, Medium, High, Critical and use those instead of 0, 1, 2, 3